ProcessLib/ProcessVariable.h

```diff
@@ -56,10 +57,21 @@ class ProcessVariable
 	/// Returns a mesh on which the process variable is defined.
 	MeshLib::Mesh const& getMesh() const;
 
-	void initializeDirichletBCs(MeshGeoToolsLib::MeshNodeSearcher& searcher,
-	        const AssemblerLib::LocalToGlobalIndexMap& dof_table,
-	        const unsigned nodal_dof_idx,
-	        std::vector<GlobalIndexType>& global_ids, std::vector<double>& values);
+	template <typename OutputIterator>
+	void initializeDirichletBCs(
+	    OutputIterator bcs,
+	    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+	    const AssemblerLib::LocalToGlobalIndexMap& dof_table,
+	    const unsigned nodal_dof_idx)
+	{
+		for (auto& bc : _dirichlet_bcs)
```

Comment:
I think, nomenclature is rather confusing: Everything is named `bc`, but has a different meaning for the output iterator than for `_dirichlet_bcs`. I think that should change in the near future.